src/plugins/intel_npu/src/plugin/npuw/serialization.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2024-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "serialization.hpp"
+
+#include "intel_npu/config/config.hpp"
+#include "logging.hpp"
+#include "openvino/op/constant.hpp"
+#include "spatial.hpp"
+
+void ov::npuw::s11n::write(std::ostream& stream, const std::streampos& var) {
+    stream.write(reinterpret_cast<const char*>(&var), sizeof var);
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const std::string& var) {
+    auto var_size = var.size();
+    stream.write(reinterpret_cast<const char*>(&var_size), sizeof var_size);
+    stream.write(&var[0], var.size());
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const bool& var) {
+    stream.write(reinterpret_cast<const char*>(&var), sizeof var);
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const ov::npuw::compiled::Spatial& var) {
+    using ov::npuw::s11n::write;
+
+    write(stream, var.params.size());
+    for (const auto& p : var.params) {
+        write(stream, p.idx);
+        write(stream, p.dim);
+    }
+    write(stream, var.range);
+    write(stream, var.nway);
+    write(stream, var.out_dim);
+    write(stream, var.nway_iters);
+    write(stream, var.tail_size);
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const ov::Tensor& var) {
+    using ov::npuw::s11n::write;
+
+    NPUW_ASSERT(var.is_continuous());
+    auto type_str = var.get_element_type().to_string();
+    write(stream, type_str);
+    write(stream, var.get_shape());
+    // FIXME: Should strides be serialized as well?
+    write(stream, var.get_byte_size());
+    stream.write(reinterpret_cast<const char*>(var.data()), var.get_byte_size());
```

Comment:
yes if you put the strided buffer.

Reply:
Well, I won't be able to deserialize it without copy than. OV only provides a constructor with strides if we also provide an external data pointer. In that case I would need to copy such tensor to another one to get the ownership. Are we OK with that?

Reply:
Done